### PR TITLE
Optional array support

### PR DESCRIPTION
**Pull request by relatko**
Support for optional public keys arguments. I see a case for refactoring the solution, there is some repeated code, however I believe, this refactoring should be done in a slightly broader scope then this pull request.

### CODE REVIEW

File: tests/parser.cpp [hunk 1]
```diff
@@ -42,10 +52,84 @@ TEST(parser, printOptional) {
...
+    EXPECT_THAT(err, PARSER_OK);
+    EXPECT_THAT(pageCountVar, 1);
+    EXPECT_STREQ(outValBuf, "None");
 
+    err = parser_printArgumentOptionalPublicKeys(&arg_list.argCtx[2], 0,
+                                               outValBuf, 40, 0, &pageCountVar);
+    EXPECT_THAT(err, PARSER_OK);
+    EXPECT_STREQ(outValBuf, "f845b8406e4f43f79d3c1d8cacb3d5f3e7aeedb");
+    EXPECT_THAT(pageCountVar, 4);
+    err = parser_printArgumentOptionalPublicKeys(&arg_list.argCtx[2], 0,
```
Comment (ppershing) [inline tests/parser.cpp:77]: can we have spaces before each separate block of test?

File: app/src/parser.c [hunk 2]
```diff
@@ -297,6 +297,48 @@ parser_error_t parser_printArgumentPublicKeys(const parser_context_t *argumentCt
...
+        if (outValLen < 5) {
+            return  PARSER_UNEXPECTED_BUFFER_END;
+        }
+        *pageCount = 1;
+        strncpy_s(outVal, "None", 5);
+    }
+    else {
+        uint16_t arrayTokenCount;
+        CHECK_PARSER_ERR(array_get_element_count(&parsedJson, internalTokenElementIdx, &arrayTokenCount));
+        if (arrayTokenCount > 64) {
```
Comment (ppershing) [inline app/src/parser.c:321]: :thinking: do we have some intuition for this check? It seems a bit arbitrary

**Review by psiemens — APPROVED** [review]
This logic looks good to me 👍

File: app/src/parser.c (hunk 2 above)
Comment (relatko) [reply app/src/parser.c:321]: Hmm. This is copy-paste from non-optional function. As far as I see, the constant 64 is really quite random there too and as of now it limits indirectly the number of Public Keys in Create Account transaction. We plan to limit this to 3 in the future. I do not want to mess with Create Account transaction now.
For now, I just introduced a new define which puts these two related constants together and I added a short comment on what the test does. This particular test will probably change very soon (when I will be confident about the state of testing of Create Account transaction).

**Pull request merged**